File: Frontend/reddit/src/app/services/comment_service/comment.service.ts
import { Injectable } from '@angular/core';
import { HttpClientModule, HttpParams, HttpClient, HttpHeaders } from '@angular/common/http';
import { Observable } from 'rxjs';
import { Comment } from 'src/app/model/comment';

@Injectable({
	providedIn: 'root'
})
export class CommentService {
	private headers = new HttpHeaders({ 'Content-Type': 'application/json' });
	private readonly baseURL = "/api/api/comment/"

	constructor(
		private http: HttpClient
	) { }

	createNew(comment: Comment): Observable<any> {
		return this.http.post(this.baseURL,
			{
				commentId: comment.commentId,
				text: comment.text,
				timestamp: new Date(),
				userId: comment.userId,
				postId: comment.postId,
				repliesToCommentId: comment.repliesToCommentId,
				deleted: false
			},
			{ headers: this.headers, responseType: 'json' });
	}

	getCommentById(commentId: number): Observable<any> {
		let queryParams = {};

		queryParams = {
			headers: this.headers,
			observe: 'response'
		};

		return this.http.get(this.baseURL + commentId, queryParams);
	}

	getAllPostComments(postId: number): Observable<any> {
		let queryParams = {};

		queryParams = {
			headers: this.headers,
			observe: 'response'
		};

		return this.http.get(this.baseURL + "post/" + postId, queryParams);
	}

	delete(commentId: number): Observable<any> {
		return this.http.delete(this.baseURL + commentId);
	  }

	getAllCommentComments(commentId: number): Observable<any> {
		let queryParams = {};

		queryParams = {
			headers: this.headers,
			observe: 'response'
		};

		return this.http.get(this.baseURL + "comment/" + commentId, queryParams);
	}

	getAllPostCommentsSortedByPopular(postId: number): Observable<any> {
		let queryParams = {};

		queryParams = {
			headers: this.headers,
			observe: 'response'
		};

		return this.http.get(this.baseURL + "post/" + postId + "/sorted-by-popular", queryParams);
	}

	getAllPostCommentsSortedByUnpopular(postId: number): Observable<any> {
		let queryParams = {};

		queryParams = {
			headers: this.headers,
			observe: 'response'
		};

		return this.http.get(this.baseURL + "post/" + postId + "/sorted-by-unpopular", queryParams);
	}

	getAllPostCommentsSortedByDateAsc(postId: number): Observable<any> {
		let queryParams = {};

		queryParams = {
			headers: this.headers,
			observe: 'response'
		};

		return this.http.get(this.baseURL + "post/" + postId + "/sorted-by-date-asc", queryParams);
	}


	getAllPostCommentsSortedByDateDesc(postId: number): Observable<any> {
		let queryParams = {};

		queryParams = {
			headers: this.headers,
			observe: 'response'
		};

		return this.http.get(this.baseURL + "post/" + postId + "/sorted-by-date-desc", queryParams);
	}

	getAllCommentCommentsSortedByPopular(commentId: number): Observable<any> {
		let queryParams = {};

		queryParams = {
			headers: this.headers,
			observe: 'response'
		};

		return this.http.get(this.baseURL + "comment/" + commentId + "/sorted-by-popular", queryParams);
	}

	getAllCommentCommentsSortedByUnpopular(commentId: number): Observable<any> {
		let queryParams = {};

		queryParams = {
			headers: this.headers,
			observe: 'response'
		};

		return this.http.get(this.baseURL + "comment/" + commentId + "/sorted-by-unpopular", queryParams);
	}

	getAllCommentCommentsSortedByDateAsc(commentId: number): Observable<any> {
		let queryParams = {};

		queryParams = {
			headers: this.headers,
			observe: 'response'
		};

		return this.http.get(this.baseURL + "comment/" + commentId + "/sorted-by-date-asc", queryParams);
	}

	getAllCommentCommentsSortedByDateDesc(commentId: number): Observable<any> {
		let queryParams = {};

		queryParams = {
			headers: this.headers,
			observe: 'response'
		};

		return this.http.get(this.baseURL + "comment/" + commentId + "/sorted-by-date-desc", queryParams);
	}

}

File: Android/Reddit/app/src/main/java/com/ognjenlazic/reddit/network/service/CommentService.kt
package com.ognjenlazic.reddit.network.service

import com.ognjenlazic.reddit.model.Comment
import com.ognjenlazic.reddit.model.Post
import com.ognjenlazic.reddit.network.model.CommentResponse
import retrofit2.http.*

interface CommentService {

    @GET("comment")
    suspend fun getAllComments(): List<Comment>

    @GET("comment/post/{postid}")
    suspend fun getAllPostComments(@Path("postid") postid: Int): List<CommentResponse>

    @GET("comment/comment/{commentId}")
    suspend fun getAllCommentComments(@Path("commentId") commentId: Int): List<CommentResponse>

    @GET("comment/{commentId}")
    suspend fun getCommentById(@Path("commentId") commentId: Int): CommentResponse

    @DELETE("comment/{commentId}")
    suspend fun deleteComment(@Path("commentId") commentId: Int)

    @POST("comment")
    suspend fun createNew(@Body comment: Comment): Comment

    @GET("comment/post/{postId}/sorted-by-popular")
    suspend fun getAllCommentsSortedByPopular(@Path("postId") postId: Int): List<CommentResponse>

    @GET("comment/post/{postId}/sorted-by-unpopular")
    suspend fun getAllCommentsSortedByUnpopular(@Path("postId") postId: Int): List<CommentResponse>

    @GET("comment/post/{postId}/sorted-by-date-desc")
    suspend fun getAllCommentsSortedByNewest(@Path("postId") postId: Int): List<CommentResponse>

    @GET("comment/post/{postId}/sorted-by-date-asc")
    suspend fun getAllCommentsSortedByOldest(@Path("postId") postId: Int): List<CommentResponse>
}